package bit.app.exception;

import java.io.File;
import java.io.FileInputStream;
import java.io.IOException;
import java.nio.channels.InterruptedByTimeoutException;

public class ExceptionEx2 {
	public void interruptedException() {
		for(int i=0; i<100; i++) {
			try {
				Thread.sleep(100);
				if(i%7==0) System.out.println();
				System.out.printf("%5d", i);
				if(i%5==0) {
					throw new InterruptedException();
				}
			} catch (InterruptedException e) {
				System.out.println("InterruptedException ...!");
				e.printStackTrace();
			}
		}
	}
	public void iOException() {
		FileInputStream fis = null;
		try {
			fis = new FileInputStream(new File("d:\\data\\ExceptionEx1.java"));
			int temp = 0;
			while((temp=fis.read())!=-1) { // bof값이 -1임
				Thread.sleep(100);
				System.out.print((char)temp);
			}
			System.out.println(System.in.read());
		} catch (IOException e) {
			e.printStackTrace();
		} catch(InterruptedException e) {
			e.printStackTrace();
		}
	}
	public static void main(String[] args) {
		ExceptionEx2 e = new ExceptionEx2();
//		e.interruptedException();
		e.iOException();
	}
}
